harness/determined/pytorch/_trainer.py

```diff
@@ -176,11 +177,39 @@ def fit(
 
             smaller_is_better = bool(self._info.trial._config["searcher"]["smaller_is_better"])
             searcher_metric_name = self._info.trial._config["searcher"]["metric"]
+
             steps_completed = int(self._info.trial._steps_completed)
             global_batch_size = self._info.trial.hparams.get("global_batch_size", None)
             if global_batch_size:
                 global_batch_size = int(global_batch_size)
 
+            # Backwards compatibility: try to parse legacy `searcher.max_length` if `max_length`
+            # isn't passed in.
+            if max_length is None:
+                max_length_val = core._parse_searcher_max_length(self._info.trial._config)
+                if max_length_val:
+                    logger.warning(
```

Comment:
please use `warnings.warning()` for deprecation warnings, grep around for examples